projects/kit/components/input-date-range/input-date-range.component.ts

```diff
@@ -157,6 +168,21 @@ export class TuiInputDateRangeComponent
         super(control, cdr, valueTransformer);
     }
 
+    @HostListener('click')
+    onClick(): void {
+        if (!this.isMobile && this.interactive) {
+            this.toggle();
+        }
+    }
+
+    // TODO: remove this after refactor controls to hold whole TuiDayRangePeriod
+    ngAfterViewChecked(): void {
+        if (this.calendarRange) {
+            this.calendarRange.selectedActivePeriod = this.selectedActivePeriod;
+            this.calendarRange.cdr.markForCheck();
```

Comment:
when calendar rerender, we need to pass saved `selectedActivePeriod` to child calendarRange displays correct active item